classes/services/queryBuilders/PKPStatsListQueryBuilder.inc.php

```diff
@@ -0,0 +1,350 @@
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterBySectionIds($sectionIds) {
+		$this->filters[STATISTICS_DIMENSION_PKP_SECTION_ID] = $sectionIds;
+		return $this;
+	}
+
+	/**
+	 * Set daily date range (from and to) filter
+	 * Assumes that either start or end date exists
+	 *
+	 * @param $dateStart string optional
+	 * @param $dateEnd string optional
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterByDateRange($dateStart = null, $dateEnd = null) {
+		// pre-assumption: one of the dates exist
+		assert($dateStart != null || $dateEnd != null);
+
+		$from = $to = null;
+		// convert the date strings to the right format
+		if ($dateStart && preg_match('/(\d{4})-(\d{2})-(\d{2})/', $dateStart, $matches) === 1) {
+			$from = $matches[1] . $matches[2] . $matches[3];
+		}
+		if ($dateEnd && preg_match('/(\d{4})-(\d{2})-(\d{2})/', $dateEnd, $matches) === 1) {
+			$to = $matches[1] . $matches[2] . $matches[3];
+		}
+		// set the default value for from and to date, if missing
+		if ($to == null) {
+			$to = date('Ymd', time());
+		} elseif ($from == null) {
+			// TO-DO eventually: chose the start date differently ?
+			$from = '20010101';
+		}
+		// set the filter
+		$this->filters[STATISTICS_DIMENSION_DAY]['from'] = $from;
+		$this->filters[STATISTICS_DIMENSION_DAY]['to'] = $to;
+	}
+
+	/**
+	 * Set filter according to the given search phrase
+	 * Assumes that lookingFor is already set
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterBySearchPhrase($searchPhrase) {
+		switch ($this->lookingFor) {
+			case 'orderedSubmissions':
+			case 'totalSubmissionsStats':
+				$submissionService = \ServicesContainer::instance()->get('submission');
+				$submissions = $submissionService->getSubmissions($this->contextId, array('searchPhrase' => $searchPhrase));
+				$submissionIds = array_map(
+					function($submission){
+						return $submission->getId();
+					},
+					$submissions
+				);
+				if (!empty($submissionIds)) {
+					$this->filterBySubmissionIds($submissionIds);
+				} else {
+					$this->noObjectForSearchPhrase = true;
+				}
+				break;
+		}
+	}
+
+	/**
+	 * Add orderBy statement
+	 *
+	 * @param $orderBy string
+	 * @param $orderDirection string
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function orderBy($orderBy, $orderDirection) {
+		$orderByColumn = $direction = null;
+		switch($orderBy) {
+			case 'total':
+				$orderByColumn = STATISTICS_METRIC;
+				break;
+			case 'month':
+				$orderByColumn = STATISTICS_DIMENSION_MONTH;
+				break;
+			case 'day':
+				$orderByColumn = STATISTICS_DIMENSION_DAY;
+				break;
+		}
+		switch($orderDirection) {
+			case 'ASC':
+				$direction = STATISTICS_ORDER_ASC;
+				break;
+			case 'DESC':
+				$direction = STATISTICS_ORDER_DESC;
+				break;
+		}
+		$this->orderBy[$orderByColumn] = $direction;
```

Comment:
Hmmm... If this get() method here, in this query builder, will eventually later be used also for other statistics reports, then I think it is possible to choose several columns to order the results by, depending on the query i.e. what one would like to have... e.g. (just imagining) sort by metric DESC and country ASC...
Currently however we only use one column. 
Do you still think I should change it so that only the current situation is represented?

Reply:
Let's simplify it for now. If we do want to order by multiple dimensions, we can add that later.